llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -2978,6 +2978,29 @@ Instruction *InstCombinerImpl::visitCallInst(CallInst &CI) {
         Intrinsic::getOrInsertDeclaration(II->getModule(), NewIntrin);
     return CallInst::Create(NewFn, CallArgs);
   }
+  case Intrinsic::ptrauth_sign: {
+    // auth + sign can be replaced with resign, which prevents unsafe
+    // spills and reloads of intermediate authenticated value.
```

Comment:
I wonder if this comment would be easier to understand by a reader who hasn't worked much on pointer authentication if it explained this in a bit more detail, for example as: `After auth, a raw, non-signed pointer resides in a register. Later compiler passes may end up spilling and reloading this raw pointer to memory, which undermines the security property that pointer authentication aims to give (not having certain classes of pointers residing unsigned/raw in memory).
By rewriting an authentication followed by a signing operation as a single resign, we eliminate the possibility of the intermediate unsigned pointer being spilled to memory`

Reply:
Will update the comment, thanks!

Reply:
Updated in 3b4d9b537dfd08941a1c4319d5758bd6ea63fae0, though, with different wording - hopefully this would be even easier to understand:
```cpp
    // Replace auth+sign with a single resign intrinsic.
    // When auth and sign operations are performed separately, later compiler
    // passes may spill intermediate result to memory as a raw, unprotected
    // pointer, which makes it possible for an attacker to replace it under
    // PAuth threat model. On the other hand, resign intrinsic is not expanded
    // until AsmPrinter, when it is emitted as a contiguous, non-attackable
    // sequence of instructions.
```

Reply:
Thanks!
After I wrote my initial comment above I wondered if this is the first peephole pattern added to InstCombine that is implemented because of security reasons rather than because of performance reasons.
Anyway thanks for that improved comment!